dashboard/app/controllers/hoc_download_controller.rb

```diff
@@ -0,0 +1,42 @@
+class HocDownloadController < ApplicationController
+
+  DOWNLOAD_URL_BASE = '//downloads.code.org/hourofcode/'
+
+  def index
+    @product_name = params[:product]
+
+    case @product_name
+      when Script::MINECRAFT_NAME
+        @app_name = t('hoc_download.minecraft_name')
+        @file_prefix = 'MC'
+        @og_image_url = CDO.code_org_url('/images/mc/mc_social.jpg')
+      when Script::STARWARS_NAME
+        @app_name = t('hoc_download.starwars_blocks_name')
+        @file_prefix = 'StarWarsBlocks'
+        @og_image_url = CDO.code_org_url('/images/star-wars-announcement.jpg')
+
+        # for JavaScript version (Star Wars only)
+        @app_name_js = t('hoc_download.starwars_javascript_name')
+        @file_prefix_js = 'StarWarsJS'
+        @show_js_links = true
+      else
+        render_404
+        return
+    end
+
+    @hoc_url = CDO.code_org_url("/#{@product_name}")
+
+    @english_windows_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "EnglishSetup.exe"
+    @english_mac_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "English.dmg"
+    @spanish_windows_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "SpanishSetup.exe"
+    @spanish_mac_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "Spanish.dmg"
```

Comment:
Yes, thx. Moved into a download_url function as you described and used string interpolation - nicer!

Agree this can be generalized into something that is not 8 similar variables. I am going to wait on that until the next phase, when we expand this to support all languages. That's when we figure out what the download UI will be, and that will affect how we'd write this. This is the most direct path to a soft beta launch for just English and Spanish.